### DISCUSSION TITLE
Level 2 Processing CC 100% but image without cloud

### DISCUSSION THREAD

**Discussion by sulmonieri** [davidfrantz/force#88] [Q&A] [answered]
I am testing the force-level2 on a single image (in summertime) without a single cloud but the image is skipped and the logfile indicates cc:100%. For the DTM, I am using a 25m resolution model (.tff) and a reprojection with a WKT string (see in attached parameter file).
I also attached the text of the .vrt file of the DTM for information as I suspect that there lies the problem.
[para.txt](https://github.com/davidfrantz/force/files/6343045/para.txt)
[swiss_dtm25.txt](https://github.com/davidfrantz/force/files/6343047/swiss_dtm25.txt)

**Comment by ernstste** [comment]
Hi there,

if you suspect your DEM to be the issue, you can run the processing without the DEM to verify your assumption.
`FILE_DEM = NULL`
As for the VRT - It's in float, @davidfrantz is that okay for force-level2?

**Reply by sulmonieri** [reply]
Indeed the problem was resolved by running without the DEM. I just found the problem. The DEM does not cover the entire tile.

**Reply by sulmonieri** [reply]
I managed to increase the size of the DTM to cover the entire tile. The same problem arose with cloud cover and then, skipping the tile. Below you see the vrt file the dtm (link) itself and the log file from the force-level2.
https://www.dropbox.com/sh/s5wlwm8b6m2ausw/AADiOE7kGWXmIWI4aGiVijTka?dl=0
[swiss_dtm25_big.txt](https://github.com/davidfrantz/force/files/6384874/swiss_dtm25_big.txt)
[S2B_MSIL1C_20200708T102559_N0209_R108_T32TMT_20200708T124214.SAFE.log](https://github.com/davidfrantz/force/files/6384878/S2B_MSIL1C_20200708T102559_N0209_R108_T32TMT_20200708T124214.SAFE.log)

**Comment by davidfrantz** [comment]
Hi @sulmonieri ,

thanks for the data. I see two issues:

the usage of the VRT is redundant as the tiff is already a mosaic. You can directly use the tiff.

I believe that the problem could be related to the nodata value. The parameter file says DEM_NODATA = -32767. In the data, the fill value is -3.40282347e+38 . I suggest to convert your DEM to INT16. Float works too, but it is not really necessary. Please burn -32767 into the DEM where there is no data, and then try again.

Hope this helps,
David

**Reply by sulmonieri** [reply]
Thanks for the quick response! I set all NODATA to -32767 and converted the DEM to INT. Unfortunately, there is another issue now displayed in the log file: DEM out of bounds: min 0.257307 max 65.536003. Elevation statistics failed! unable to compile topography.
https://www.dropbox.com/s/fcuxwww709sc4q4/DTM25_large_int.tif?dl=0
I don't understand, the limits of the DEM should be far larger than 0.257 - 65.536

**Comment by davidfrantz** [comment, marked as answer]
This message prints the range in kilometers. Thus, you still have some weird values in the DEM (65536), or potentially used Unsigned Int, in which case the negative values (-32767) might got wrapped around to the upper UInt16 range...

**Reply by sulmonieri** [reply]
Thanks! Indeed the problem was with unsigned Int.
